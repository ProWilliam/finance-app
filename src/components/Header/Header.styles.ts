import React, { useState } from 'react';
import { StyleSheet } from 'react-native';
import theme from '../../theme/theme';

const StyleHeader = StyleSheet.create({
  container: {
    width: `${theme.withScreen.big}%`,
    borderBottomWidth: 2, 
    paddingBottom: 10,
    marginBottom: 20,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: "center",
    borderBottomColor: theme.light.border,
    backgroundColor: theme.light.backgroundPrimary, 

  },
  icon: {
    marginRight: 10,
    color: theme.light.icon,
    fontSize: theme.size.iconMedium,
  },
  title: {
    fontSize: theme.size.fontMedium,
    fontWeight: 'bold',
    justifyContent: 'flex-start'
  }
});

export default StyleHeader;