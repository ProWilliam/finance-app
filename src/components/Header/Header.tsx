import React from 'react';
import { View, Text } from 'react-native';

// Component
import IconBank from '../../assets/icons/IconBank';

// Styles
import styles from './Header.styles';


const Header = () => {
  return(
    <View style={styles.container}>
      <IconBank style={styles.icon} />
      <Text style={styles.title}>
        Banco
      </Text>
    </View>
  )
}
export default Header;