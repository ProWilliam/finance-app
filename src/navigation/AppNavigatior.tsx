import React from 'react';
import { View } from 'react-native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

// Components
import Header from '../components/Header/Header';
import FinanceProductsScreen from '../screens/Products/FinanceProductsScreen';
import InfoProductScreen from '../screens/InfoProduct/InfoProductScreen';
import RegisterFormScreen from '../screens/RegisterForm/RegisterFormScreen'; 

// Styles
import styleMain from './AppNavigatior.styles';

const Stack = createNativeStackNavigator();

const AppNavigatior: React.FC = () => {

  return (
    <View style={styleMain.container}>
      <Header />
      <Stack.Navigator initialRouteName="Home"> 
        <Stack.Screen name="Home" component={FinanceProductsScreen} options={{headerShown: false}} />
        <Stack.Screen name="Info" initialParams={{id: ''}} component={InfoProductScreen} options={{headerShown: false}} /> 
        <Stack.Screen name="AddProduct" component={RegisterFormScreen} options={{headerShown: false}} /> 
      </Stack.Navigator>
    </View>
  )
}

export default AppNavigatior;