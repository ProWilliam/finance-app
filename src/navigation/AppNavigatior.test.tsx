import React from 'react';
import renderer, { act, ReactTestRenderer } from 'react-test-renderer';
import { View } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

import AppNavigatior from './AppNavigatior'; 
import Header from '../components/Header/Header';

// Mocks for components
jest.mock('../components/Header/Header', () => 'Header');
jest.mock('../screens/Products/FinanceProductsScreen', () => 'FinanceProductsScreen');
jest.mock('../screens/InfoProduct/InfoProductScreen', () => 'InfoProductScreen');
jest.mock('../screens/RegisterForm/RegisterFormScreen', () => 'RegisterFormScreen');
jest.mock('../screens/EditProduct/EditProductScreen', () => 'EditProductScreen');

// Mock React Navigation modules if necessary
jest.mock('@react-navigation/native', () => {
  const actual = jest.requireActual('@react-navigation/native');
  return {
    ...actual,
    useNavigation: jest.fn().mockReturnValue({
      navigate: jest.fn(),
      goBack: jest.fn(),
    }),
  };
});

describe('<AppNavigatior />', () => {
  let tree: ReactTestRenderer;


  afterEach(() => {
    jest.clearAllMocks
  })

  it('should render correctly', async () => {
    
    await act(() => {
      tree = renderer.create(
        <NavigationContainer>
          <AppNavigatior />
        </NavigationContainer>
      );
    })
    
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render Header component', async () => {
    
    const root = tree.root;
    
    // Verify that the Header component is rendered
    expect(root.findByType(Header)).toBeDefined();
  });

  it('should contain Stack.Navigator', () => {
    
    const root = tree.root;
    
    // Verifica que el Stack.Navigator estÃ© renderizado
    const navigator = root.findByType(View).findByType(createNativeStackNavigator().Navigator);
    expect(navigator).toBeDefined();
  });

  it('should have initial route name as Home', () => {
    
    const root = tree.root;
    
    // Verifica que el Stack.Navigator tenga el nombre de ruta inicial correcto
    const navigator = root.findByType(createNativeStackNavigator().Navigator);
    expect(navigator.props.initialRouteName).toBe('Home');
  });

});
