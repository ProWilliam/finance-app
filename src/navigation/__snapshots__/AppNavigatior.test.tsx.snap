// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppNavigatior /> should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
      "marginTop": 54,
    }
  }
>
  <Header />
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        hasLargeHeader={false}
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onGestureCancel={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillAppear={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        sheetAllowedDetents="large"
        sheetCornerRadius={-1}
        sheetExpandsWhenScrolledToEdge={true}
        sheetGrabberVisible={false}
        sheetLargestUndimmedDetent="all"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            collapsable={false}
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <FinanceProductsScreen
              navigation={
                {
                  "addListener": [Function],
                  "canGoBack": [Function],
                  "dispatch": [Function],
                  "getId": [Function],
                  "getParent": [Function],
                  "getState": [Function],
                  "goBack": [Function],
                  "isFocused": [Function],
                  "navigate": [Function],
                  "pop": [Function],
                  "popToTop": [Function],
                  "push": [Function],
                  "removeListener": [Function],
                  "replace": [Function],
                  "reset": [Function],
                  "setOptions": [Function],
                  "setParams": [Function],
                }
              }
              route={
                {
                  "key": "Home-xZ8fA8zF-GLWICFPzhX0P",
                  "name": "Home",
                  "params": undefined,
                }
              }
            />
          </View>
        </View>
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backTitleVisible={true}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={true}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="Home"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
      </RNSScreen>
    </RNSScreenStack>
  </RNCSafeAreaProvider>
</View>
`;
